src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java

```diff
@@ -143,4 +171,35 @@ public AsyncJsonTokenParser(
     return Futures.transformAsync(futureVerifiers, checkNullFunction, executor);
   }
 
+  /**
+   * Converts exceptions (when applicable) to {@link InvalidJsonTokenException}
+   * for better exception handling in the asynchronous parser.
+   *
+   * @return a failed {@link ListenableFuture}
+   */
+  private ListenableFuture getFailedFuture(Exception e) {
```

Comment:
There are additional public methods exposed by AsyncJsonTokenParser (inherited from AbstractJsonTokenParser). Can we map the exceptions for those methods too?

Reply:
We can, but we would need to make InvalidJsonTokenException a runtime exception. Otherwise, we would need to change the method signatures of existing public methods, which wouldn't be possible, to indicate that it is thrown.

Reply:
Or wrap it in a runtime exception when rethrown in an existing public method.

Reply:
We could probably change the method signature of only the async interface by making the methods in the abstract class protected and creating new public methods with the new signature.

Reply:
Since changing the function signature by adding a checked exception conflicts with the old function, I mostly split the public methods. Some of the previously public methods from the abstract parser that we knew to throw exceptions (verify, deserialize, signatureIsValid) are now renamed package-private methods (abstractVerify, abstractDeserialize, abstractSignatureIsValid) in the abstract parser that both the sync and async parser will call in their public functions. This allows the other public function in the async parser to have checked exceptions.